src/lib/mina_generators/zkapp_command_generators.ml

```diff
@@ -1843,4 +1843,59 @@ let%test_module _ =
             ~random:(Splittable_random.State.create Random.State.default))
       in
       ()
+
+    let%test_unit "generate max cost zkapp command" =
+      let keypair = Keypair.create () in
+      let public_key_compressed =
+        Signature_lib.Public_key.compress keypair.public_key
+      in
+
+      let account_state_tbl = Account_id.Table.create () in
+      let fee_payer_account_id =
+        Account_id.create public_key_compressed Token_id.default
+      in
+      let fee_payer_account =
+        Account.create fee_payer_account_id
+          (Currency.Balance.of_mina_int_exn 1000)
+      in
+      Account_id.Table.set account_state_tbl ~key:fee_payer_account_id
+        ~data:(fee_payer_account, `Fee_payer) ;
+
+      for _i = 1 to 10 do
+        let zkapp_keypair = Signature_lib.Keypair.create () in
+        let zkapp_pk =
+          Signature_lib.Public_key.compress zkapp_keypair.public_key
+        in
+        let zkapp_account_id = Account_id.create zkapp_pk Token_id.default in
```

Comment:
Could we consider moving parts of the logics here into `Mina_base.Account.gen`, add an optional arg `?public_key` to override pub key if needed?